e2e/openshift/openshift_packaging_test.go

```diff
@@ -0,0 +1,340 @@
+
+	// create a YAMLTemplate for the go templating
+	yaml := &YAMLTemplate{
+		AppVersion:     appVersion,
+		DockerRegistry: dockerRegistry,
+	}
+
+	// create some YAML files
+	catalogByte := parseTmpl(t, catalogSource, yaml)
+	subByte := parseTmpl(t, subscription, yaml)
+
+	catalogFilename := writeFile(t, catalogByte)
+	subFilename := writeFile(t, subByte)
+
+	var b bytes.Buffer
+	b.WriteString(operatorGroup)
+
+	operatorFilename := writeFile(t, b.Bytes())
+
+	files := []string{catalogFilename, subFilename, operatorFilename}
+
+	// install YAML files which install the operator
+	for _, f := range files {
+		fmt.Println(f)
+		args := []string{"delete", "-f", f}
+		process.ExecJUnit("oc", args, os.Environ())
+		args = []string{"create", "-f", f}
+		require.NoError(t, process.ExecJUnit("oc", args, os.Environ()), "failed creating openshift operator files")
+	}
+
+	cs, err := createClientset()
+	require.NoError(t, err, "failed to create clientset")
+
+	// test that the operator pod is up and running
+	// this test checks that all pods are running in a ns
+	// but that works.
+	require.NoError(t, testPods(t, cs), "failed finding operator")
+
+	// create an example database
+	args = []string{
+		"create", "-f", "../../examples/example.yaml",
+	}
+
+	require.NoError(t, process.ExecJUnit("oc", args, os.Environ()), "failed creating crdb database")
+
+	testLog := zapr.NewLogger(zaptest.NewLogger(t))
+
+	// wait till the sts is ready, no the code that we have actually does not fully work
+	// and no other code is using it, and there is an issue open. It returns
+	// before all of the pods are running.
+	require.NoError(t,
+		kube.WaitUntilAllStsPodsAreReady(
+			context.TODO(),
+			cs,
+			testLog,
+			"cockroachdb",
+			"test",
+			800*time.Second,
+			10*time.Second,
+		),
+		"crdb database did not start",
+	)
+
+	// testing that all pods are running in the namespace, except for the job pod
+	// this will check that the pods are up and going
+	require.NoError(t, testPods(t, cs))
+
+	t.Log("rejoice and celebrate as our operator and OpenShift are happy")
+
+	// TODO: We do not do any cleanup at this time, but the test will run again if we need it to.
+	// We could write our own require.NoError and then remove the YAML files and the test
+	// directory as well to clean house a bit.
+}
+
+// letters are used to generate a random string of letters by randSeq
+var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+// randSeq returns a string of n letters
+func randSeq(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+// wriiteFile writes out b byte array to a random file in the
+// os.TempDir and returns the name of that file.
+func writeFile(t *testing.T, b []byte) string {
+	rand.Seed(time.Now().UnixNano())
+	fileName := fmt.Sprintf("%s-test.yaml", randSeq(10))
+	fileName = filepath.Join(os.TempDir(), fileName)
```

Comment:
Did you want to use `t.TempDir()` instead so that there's no artifacts left behind?

Reply:
Ohhhhh! Perfect. Did not know that trick!